HEN_HOUSE/egs++/ausgab_objects/egs_phsp_scoring/egs_phsp_scoring.h

```diff
@@ -0,0 +1,266 @@
+          return ~( (1 << 31) | (1 << 30) | (1 << 29) );
+    }
+    static unsigned int bsqe() {
+           return (1 << 30);
+    }
+    static unsigned int bsqp() {
+           return (1 << 29);
+    }
+    struct egs_phsp_write_struct {
+       int   latch;
+       float E;
+       float x,y;
+       float u,v;
+       float wt;
+       egs_phsp_write_struct() {};
+       egs_phsp_write_struct(const Particle &p) {
+        latch = (p.latch & bclr());
+        if( p.q == -1 ) latch = (latch | bsqe());
+        else if( p.q == 1 ) latch = (latch | bsqp());
+        E = p.E; x = p.x; y = p.y; u = p.u; v = p.v;
+        wt = p.w >= 0 ? p.wt : -p.wt;
+       };
+    };
+    mutable fstream phsp_file; //output file -- mutable so we can write to it during storeState
+    EGS_I64 count; //total no. of particles in file
+    EGS_I64 countg; //no. of photons in file
+    float emax; //max. k.e. of particles in phsp file
+    float emin; //min. k.e. of charged particles in file
+
+    //variables specific to IAEA format
+    mutable int iaea_id; //file id--mutable so we can write to it during storeState
+
+    int latch_ind; //IAEA id for latch variable (set to 2)
+    int iaea_n_extra_float, iaea_n_extra_long; //no. of extra floats, ints
+    int iaea_i_latch, iaea_i_muindex; //indices of indicated variables in arrays...may not need these
+    int iaea_q_type[3]; //easy conversion from q to iaea type
+    bool xyz_is_const[3]; //set to true if scoring at constant X/Y/Z
+    float xyzscore[3]; //constant X/Y/Z scoring values
+    const string xyzname[3] = {"X", "Y", "Z"};
+    int len; //length of name
+    char *phsp_fname_char; //need file name in char format
+
+    //back to variables common to EGSnrc and IAEA formats
+
+    Particle *p_stack; //the stored particle stack
+
+    bool phsp_before; //true if inside scoring geometry before step
+    bool phsp_after;  //true if inside scoring geometry after step
+
+    int store_max; //max. no. of particles to store in p_stack
+    mutable int phsp_index; //index in p_stack array -- mutable so we can change it in storeState
+    mutable bool first_flush; //first time writing to file in this run -- mutable so we can change it in storeState
+
+    bool is_restart; //true if this is a restart
+    EGS_I64 countprev; //no. of particles written to file before restart
+
+    EGS_I64    last_case;   //last primary history scored
+    EGS_I64    current_case; //current primary history
+
+    EGS_BaseGeometry *phsp_geom; //geometry on entrance to/exit from which phase space data is scored
+    int oformat;           //0 for EGSnrc format, 1 for IAEA format
+
+    int ocharge;           //particle type for output: 0--all; 1--photons; 2--charged particles
+
+    int scoredir;           //scoring direction: 0--on entry and exit; 1--on entry; 2--on exit
+
+    string phsp_fname; //name of phase space file
+
+    string phspoutdir; //output directory
+
+    const EGS_Float prm = 0.5109989461; //precise rest mass
```

Comment:
This was added a while ago! See `egs_phsp_source.cpp`.
```
EGS_Application::activeApplication()->getRM()
```

Reply:
I figured there was something like that.  Thanks, Reid!  Other changes coming, too.